os/include/tinyara/userspace.h

```diff
@@ -80,8 +80,13 @@ extern uint32_t __uflash_segment_start__[];
 /* This macro is used to access the struct userpace_s header that can be
  * found at the beginning of the user-space blob.
  */
+#ifdef CONFIG_SUPPORT_COMMON_BINARY
+struct userspace_s;
```

Comment:
Is this needed?

Reply:
How about switching them?
```
struct userspace_s {
...
};

#ifdef CONFIG_SUPPORT_COMMON_BINARY
extern struct userspace_s *USERSPACE;
#else
#define USERSPACE ((FAR struct userspace_s *)__uflash_segment_start__)
#endif 
```

Reply:
And originally `#define USERSPACE ((FAR struct userspace_s *)__uflash_segment_start__)` is defined at previous of structure definition without prototype. It means that new definition should be ok. But anyway I think switching is better.

Reply:
Ok. done.